src/weapon/w_006.c

```diff
@@ -62,11 +66,120 @@ static void EntityWeaponAttack(Entity* self) {
     self->rotPivotY = PLAYER.rotPivotY;
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_006", func_ptr_80170004);
+static void func_ptr_80170004(Entity* self) {
+    s32 hitboxSize;
+    u16 data;
+
+    switch (self->step) {
+    case 0:
+        self->posY.i.hi += PLAYER.hitboxOffY;
+        if (PLAYER.facingLeft != 0) {
+            self->posX.i.hi = self->posX.i.hi - PLAYER.hitboxOffX;
+        } else {
+            self->posX.i.hi = self->posX.i.hi + PLAYER.hitboxOffX;
+        }
+        SetSpriteBank1(g_Animset);
+        self->animSet = ANIMSET_OVL(0x10);
+        self->palette = 0x116;
+        self->unk5A = 0x64;
+        if (g_HandId != 0) {
+            self->animSet += 2;
+            self->palette += 0x18;
+            self->unk5A += 2;
+        }
+        self->animCurFrame = 0x42;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->drawFlags = DRAW_COLORS | DRAW_UNK02 | DRAW_TRANSP;
+        self->rotX = self->rotY = 0x40;
+        self->ext.weapon.unk80 = 0x40;
+
+        SetSpeedX(rand() - FIX(4));
+
+        if (self->facingLeft == 0) {
+            self->ext.weapon_006.unk84 = FIX(15.0 / 64);
```

Comment:
Why are we mixing weapon and weapon_006?

Reply:
Whoops, fixed.